refactor: update root file_picker package and example app for federated architecture (#2110)
miguelpruivo/flutter_file_picker: README.md

</p>

# File Picker

> A federated plugin that allows you to use the native file explorer to pick single or multiple files, with extensions filtering support.

Reviewer: file_picker being federated is an implementation detail

```suggestion
A plugin that allows you to use the native file explorer to pick single or multiple files, with extensions filtering support.
```

## Currently supported features
* Uses OS default native pickers

> * Supports multiple platforms (Mobile, Web, Desktop) via federated architecture

Reviewer: ```suggestion
* Supports multiple platforms (Mobile, Web, Desktop)
```

### Darwin implementation notes

The iOS and macOS native implementations live under the shared Darwin source tree (`file_picker_darwin`). The iOS implementation requires iOS 14.0 or newer because it uses `PHPickerViewController` and `PHPickerResult`.

## Migrating to v12

Version 12.0 transitions `file_picker` to a **federated plugin architecture**.

### Key Breaking Changes & Migration Steps

> 1. **`FilePicker.pickFiles()` Returns `List<PlatformFile>`**:

Reviewer: Should this be moved to the Wiki?

Author: I agree that many developers overlook the GitHub Wiki and land straight on `pub.dev` or the `README.md` when upgrading packages (pub.dev doesn't surface the Wiki at all). 

Having a concise migration section directly in `README.md` provides immediate visibility for developers upgrading to v12. Long-term, building a dedicated documentation site (via GitHub Pages/Docusaurus) would be awesome, but for now, keeping this summary in the README ensures nobody misses these breaking changes when upgrading.

   - `FilePickerResult` has been removed in favor of direct lists of `PlatformFile`.
   - Returns an empty list (`[]`) if the user canceled the operation.
   - **v11**: `FilePickerResult? result = await FilePicker.pickFiles();`
   - **v12**: `List<PlatformFile> files = await FilePicker.pickFiles();`

2. **Single File Picking**:
   - Use `FilePicker.pickFile()` to pick a single file returning `PlatformFile?`.

3. **Reading Bytes and Streaming**:
   - Instead of using `withData: true` or `withReadStream: true` flags, use `PlatformFile` methods directly:
     - `Uint8List bytes = await file.readAsBytes();`
     - `Stream<Uint8List> stream = file.readAsByteStream();`

4. **Platform Options**:

>    - Platform-specific parameters are grouped into options objects:

Reviewer: ```suggestion
   - Platform-specific parameters are grouped into configuration options, with implementations per platform:
```
